src/vocab.rs

```diff
@@ -83,13 +137,53 @@ pub enum RDF {
     Type,
 }
 
+impl TryFrom<&str> for RDF {
```

Comment:
I think you should just delegate to the `TryFrom<[u8]>` implementation here:
```rust
impl TryFrom<&str> for Namespace {
    type Error = HornedError;

    fn try_from(value: &str) -> Result<Self, Self::Error> {
        Self::try_from(value.as_bytes())
    }
}
```

And also you could implement `FromStr` if you're gonna implement `TryFrom<&str>`.